src/three-components/ModelScene.ts

```diff
@@ -216,31 +219,36 @@ export default class ModelScene extends Scene {
    * Called when the model's contents have loaded, or changed.
    */
   onModelLoad(event: {url: string}) {
-    this.updateStaticShadow();
+    this.setShadowIntensity(this.shadowIntensity);
+    this.element[$needsRender]();
     this.dispatchEvent({type: 'model-load', url: event.url});
   }
 
   /**
-   * Called to update the shadow rendering when the model changes.
+   * Sets the shadow's intensity, lazily creating the shadow as necessary.
    */
-  updateStaticShadow() {
-    if (!this.model.hasModel() || this.model.size.length() === 0) {
-      this.pivot.remove(this.shadow);
-      return;
+  setShadowIntensity(shadowIntensity: number) {
+    this.shadowIntensity = shadowIntensity;
+    if (shadowIntensity > 0 && this.model.hasModel()) {
+      if (this.shadow == null) {
+        this.shadow = new Shadow(this);
+        // this.showShadowHelper();
+      } else {
+        this.shadow.setScene(this);
+      }
+      this.shadow.setIntensity(shadowIntensity);
     }
+  }
 
-    // Remove and cache the current pivot rotation so that the shadow's
-    // capture is unrotated so it can be freely rotated when applied
-    // as a texture.
-    const currentRotation = this.pivot.rotation.y;
-    this.setPivotRotation(0);
-
-    this.shadow.render(this.renderer.renderer, this);
-
-    // Lazily add the shadow so we're only displaying it once it has
-    // a generated texture.
-    this.pivot.add(this.shadow);
-    this.setPivotRotation(currentRotation);
+  /**
+   * Renders a box representing the shadow camera, which is helpful in
+   * debugging.
+   */
+  showShadowHelper() {
```

Comment:
My two cents on this: seems helpful, but I would comment it out (including the `import` that pulls in the class) because it will have a negative impact on tree-shaking when bundling.